strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/remote/heartbeat/RemoteRepositoryAlivenessCacheManagerTest.java

```diff
@@ -0,0 +1,157 @@
+package org.carlspring.strongbox.storage.repository.remote.heartbeat;
+
+import org.carlspring.strongbox.config.*;
+import org.carlspring.strongbox.config.hazelcast.HazelcastConfiguration;
+import org.carlspring.strongbox.config.hazelcast.HazelcastInstanceId;
+import org.carlspring.strongbox.data.CacheManagerTestExecutionListener;
+import org.carlspring.strongbox.data.CacheName;
+import org.carlspring.strongbox.storage.repository.remote.RemoteRepository;
+import org.carlspring.strongbox.testing.NullLayoutConfiguration;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.*;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestExecutionListeners;
+
+import javax.inject.Inject;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+@ActiveProfiles({ "test",
+                  "RemoteRepositoryAlivenessCacheManagerTestConfig" })
+@TestExecutionListeners(listeners = { CacheManagerTestExecutionListener.class },
+                        mergeMode = TestExecutionListeners.MergeMode.MERGE_WITH_DEFAULTS)
+public class RemoteRepositoryAlivenessCacheManagerTest
+{
+
+    @Inject
+    private CacheManager cacheManager;
+
+    private RemoteRepositoryAlivenessCacheManager remoteRepositoryAlivenessCacheManager;
+
+    private final static String REMOTE_REPOSITORY_URL = "http://mocked-remote-repository-url";
+
+    @BeforeEach
+    void setUp()
+    {
+        remoteRepositoryAlivenessCacheManager = new RemoteRepositoryAlivenessCacheManager(cacheManager);
+    }
+
+    @Test
+    public void isAliveShouldReturnTrueWhenRemoteRepositoryNotCached()
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        boolean alive = remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository);
+
+        assertTrue(alive);
+    }
+
+    @Test
+    public void isAliveShouldReturnTrueWhenRemoteRepositoryCachedValueIsTrue()
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        initializeCache(true);
+
+        boolean alive = remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository);
+
+        assertTrue(alive);
+    }
+
+    @Test
+    public void isAliveShouldReturnFalseWhenRemoteRepositoryCachedValueIsFalse()
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        initializeCache(false);
```

Comment:
seems this `initializeCache` is excess because of below  `remoteRepositoryAlivenessCacheManager.put`

Reply:
I was debating a bit whether to just use `remoteRepositoryAlivenessCacheManager.put` but decided against it to avoid risking the possibility of a bug in `remoteRepositoryAlivenessCacheManager.put` affecting the other tests. We're going to be testing `remoteRepositoryAlivenessCacheManager.put` so that's not a valid concern. I'll remove `initializeCache`.

Reply:
Having looked this over again I'm actually leaning toward keeping `initializeCache` and removing the call to `remoteRepositoryAlivenessCacheManager.put` in this test.